src/components/DiffView/index.tsx

```diff
@@ -12,35 +13,66 @@ import {
 import { withRouter, RouteComponentProps } from 'react-router-dom';
 import makeClassName from 'classnames';
 
+import { ApplicationState, ConnectedReduxProps } from '../../configureStore';
+import LinterMessage from '../LinterMessage';
 import refractor from '../../refractor';
+import {
+  LinterMessagesByPath,
+  fetchLinterMessages,
+  selectMessageMap,
+} from '../../reducers/linter';
+import { Version } from '../../reducers/versions';
 import { getLanguageFromMimeType, gettext } from '../../utils';
 import styles from './styles.module.scss';
 import 'react-diff-view/style/index.css';
 
+type LoadData = () => void;
+
 export type PublicProps = {
+  _loadData?: LoadData;
   diffs: DiffInfo[];
   mimeType: string;
+  version: Version;
 };
 
 export type DefaultProps = {
   _document: typeof document;
+  _fetchLinterMessages: typeof fetchLinterMessages;
   _tokenize: typeof tokenize;
   viewType: DiffProps['viewType'];
 };
 
+type PropsFromState = {
+  linterMessages: LinterMessagesByPath | null | void;
+  linterMessagesAreLoading: boolean;
+};
+
 export type RouterProps = RouteComponentProps<{}>;
 
-type Props = PublicProps & DefaultProps & RouterProps;
+export type Props = PublicProps &
+  DefaultProps &
+  PropsFromState &
+  RouterProps &
+  ConnectedReduxProps;
 
 export class DiffViewBase extends React.Component<Props> {
+  loadData: LoadData;
+
   static defaultProps: DefaultProps = {
     _document: document,
+    _fetchLinterMessages: fetchLinterMessages,
     _tokenize: tokenize,
     viewType: 'unified',
   };
 
+  constructor(props: Props) {
+    super(props);
+    this.loadData = props._loadData || this._loadData;
```

Comment:
Is all of this used to "fix" https://github.com/mozilla/addons-code-manager/issues/368? I am not a huge fan of polluting the public API that way. Could we try to stub `dispatch` instead?

Reply:
My intention was to use dependency injection to make an assertion that `componentDidMount()` and `componentDidUpdate()` will run the exact same loader function. The assertions are made [here](https://github.com/mozilla/addons-code-manager/blob/00b49f9cb33a650f01536aa925b281aac4012a9b/src/components/DiffView/index.spec.tsx#L263-L281). This approach explicitly checks that the same loader is used whereas stubbing `dispatch()` would only implicitly check it. 

An alternative would be to duplicate all load logic tests, the first set using `componentDidMount()` and the second using `componentDidUpdate()`. This seems like an unecessary burden.

It's a bit clunky to do dependency injection like this (although we do it for thunks, too). Once we port this component to `useEffect()`, it can go away.

Reply:
ah, I see. Could you please add a comment above the `loadData` variable declaration?